src/views/docs-view/utils/get-back-to-link.ts

```diff
@@ -21,27 +22,21 @@ export function getBackToLink(
 	product: Pick<ProductData, 'name' | 'slug' | 'rootDocsPaths'>
 ): SidebarBackToLinkProps {
 	/**
-	 * A rootDocsPath is considered "nested" if it has multiple path parts.
+	 * Handle "nested" rootDocsPaths, for which we want to link back
+	 * to the parent rootDocsPath. For example, "/terraform/plugin/framework"
+	 * links back to "/terraform/plugin".
 	 */
-	const pathParts = currentRootDocsPath.path.split('/')
-	const isNestedPath = pathParts.length > 1
-	/**
-	 * If we have a nested path, we try to find the parent rootDocsPath.
-	 * Note that we may not find a match.
-	 */
-	if (isNestedPath) {
-		const parentPath = pathParts.slice(0, pathParts.length - 1).join('/')
-		const parentMatches = product.rootDocsPaths.filter((rootDocsPath) => {
-			return rootDocsPath.path == parentPath
-		})
-		if (parentMatches.length > 0) {
-			const parentDocsPath = parentMatches[0]
-			return {
-				text: parentDocsPath.name,
-				href: `/${product.slug}/${parentDocsPath.path}`,
-			}
+	const parentRootDocsPath = getParentRootDocsPath(
```

Comment:
Relatively minor refactor here to use the shared utility I extracted. Should function exactly the same, and we have a slight bit more confidence here thanks to the unit tests I was previously compelled to write 🙌 😝 thanks for the reminder Bryce 🙇‍♂️